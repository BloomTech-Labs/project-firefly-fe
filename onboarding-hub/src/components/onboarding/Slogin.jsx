import React, { Component } from 'react'
import axios from 'axios'
import { signInWithGoogle } from '../../firebase/firebase.utils.js'
import { doFacebookSignIn } from '../../firebase/firebase.utils.js'

//import 'bootstrap/dist/css/bootstrap.css'
import './onboarding-css/Slogin.css'

export default class Slogin extends Component {
	state = {
		credentials: {
			email: '',
			password: '',
			passwordCheck: ''
		}
	}

	handleChange = e => {
		this.setState({
			credentials: {
				...this.state.credentials,
				[e.target.name]: e.target.value
			}
		})
		console.log(this.state.credentials)
	}

	onSubmit = e => {
		e.preventDefault()
		console.log('submit', this.state.credentials)

		axios
			.post('#', this.state.credentials)
			.then(res => {
				console.log(res)
			})
			.catch(err => {
				console.log(err)
			})
	}
	componentDidMount() { }

	render() {
		return (
			<div id='login-box'>
				<div class='left'>
					<h1 className='signup'>SIGN UP</h1>

					{/* <input type='text' name='username' placeholder='Username' /> */}
					<input
						type='email'
						name='email'
						placeholder='E-mail'
						value={this.state.credentials.email}
						onChange={this.handleChange}
					/>
					<p className='promise'>
						We will never share your credentials with anyone!
					</p>

					<input
						type='password'
						name='password'
						placeholder='Password'
						value={this.state.credentials.password}
						onChange={this.handleChange}
					/>

					<input
						type='password'
						name='passwordCheck'
						placeholder='Retype password'
						value={this.state.credentials.passwordCheck}
						onChange={this.handleChange}
					/>

					<button
						className='push_button green'
						type='submit'
						name='signup_submit'
						value='Sign me down'
					>
						Sign Me Up</button>
					<label for='button'></label>
					<p className='log-footer'>
						Already have an account ? <a href='./Signin'>Click Here</a>
					</p>
				</div>

				<div class='right flex'>
					<div className='test'>
						<button onClick={doFacebookSignIn} class='social-signin facebook'>
							LOG IN WITH FACEBOOK
						</button>
						<button onClick={signInWithGoogle} class='social-signin google'>
							LOG IN WITH GOOGLE
						</button>
					</div>
					<span className='loginwith'>{/* <br /> */}</span>
				</div>
				<div class='or'>OR</div>

			</div>
		)
	}
}
