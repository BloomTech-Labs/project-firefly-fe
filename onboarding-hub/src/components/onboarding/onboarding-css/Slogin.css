#login-box {
	position: relative;
	margin: 10% auto;
	width: 600px;
	height: 400px;
	background: #fff;
	border-radius: 2px;
	box-shadow: 0 2px 4px rgba(0, 0, 0, 0.4);
}

.left {
	/* position: absolute;
	top: 0;
	left: 0; */
	display: flex;
	flex-flow: column nowrap;
	justify-content: space-between;
	align-items: center;
	box-sizing: border-box;
	/* margin-top: 15px; */
	padding-top: 15px;
	width: 300px;
	height: 400px;
}

.signup {
	/* margin: 0 0 20px 0; */
	font-weight: 900;
	font-size: 2.4rem;
	font-family: "Nunito", cursive;
}

input[type='text'], input[type='email'], input[type='password'] {
	display: block;
	box-sizing: border-box;
	margin-bottom: 20px;
	padding: 4px;
	width: 220px;
	height: 32px;
	border: none;
	border-bottom: 1px solid #aaa;
	font-family: 'Roboto', sans-serif;
	font-weight: 400;
	font-size: 15px;
	transition: 0.2s ease;
}

input[type='text']:focus, input[type='email']:focus, input[type='password']:focus {
	border-bottom: 2px solid #16a085;
	color: #16a085;
	transition: 0.2s ease;
}

input[type='submit'] {
	width: 120px;
	height: 32px;
	background: #16a085;
	border: none;
	border-radius: 2px;
	color: #fff;
	font-family: 'Roboto', sans-serif;
	font-weight: 500;
	text-transform: uppercase;
	transition: 0.1s ease;
	cursor: pointer;
}

input[type='submit']:hover, input[type='submit']:focus {
	opacity: 0.8;
	box-shadow: 0 2px 4px rgba(0, 0, 0, 0.4);
	transition: 0.1s ease;
}

input[type='submit']:active {
	opacity: 1;
	box-shadow: 0 1px 2px rgba(0, 0, 0, 0.4);
	transition: 0.1s ease;
}

.or {
	position: absolute;
	top: 180px;
	left: 280px;
	width: 40px;
	height: 40px;
	background: #ddd;
	border-radius: 50%;
	box-shadow: 0 2px 4px rgba(0, 0, 0, 0.4);
	line-height: 40px;
	text-align: center;
}

.right {
	position: absolute;
	top: 0;
	right: 0;
	box-sizing: border-box;
	padding: 40px;
	width: 300px;
	height: 400px;
	background-image: url('../../../assets/coolfly.png');
	background-size: cover;
	background-position: center;
	border-radius: 0 2px 2px 0;
}

.right .loginwith {
	display: block;
	margin-bottom: 40px;
	font-size: 28px;
	color: #4aa810;
	text-align: center;
}

.flex {
	display: flex;
	flex-flow: column;
	justify-content: flex-end;
	align-items: flex-end;
}

.test {
	margin-bottom: -10px;
}

.loginwith {
	position: absolute;
	bottom: 0;
	margin-right: 1.6rem;
	color: green;
	text-align: center;
}

.log-footer {
	text-align: center;
	font-size: .85rem;
	width: 100%;
}

button.social-signin {
	width: 220px;
	height: 36px;
	padding: 5px;
	margin-top: 6px;
	font-size: 0.8em;
	border: none;
	border-radius: 8px;
	color: #fff;
	font-family: 'Nunito', sans-serif;
	font-weight: 900;
	transition: 0.2s ease;
	cursor: pointer;
}

.promise {
	font-size: 0.6em;
	font-weight: bold;
	margin-top: -1.5em;
}

.push_button {
	position: relative;
	font-size: 1.5em;
	text-transform: uppercase;
	width: 220px;
	height: 50px;
	text-align: center;
	color: #FFF;
	text-decoration: none;
	line-height: 43px;
	font-family: 'Oswald', Helvetica;
	display: block;
}

.push_button:before {
	background: #f0f0f0;
	background-image: -webkit-gradient(linear, 0% 0%, 0% 100%, from(#D0D0D0), to(#f0f0f0));
	border-radius: 5px;
	-webkit-box-shadow: 0 1px 2px rgba(0, 0, 0, .5) inset, 0 1px 0 #FFF;
	-moz-box-shadow: 0 1px 2px rgba(0, 0, 0, .5) inset, 0 1px 0 #FFF;
	box-shadow: 0 1px 2px rgba(0, 0, 0, .5) inset, 0 1px 0 #FFF;
	position: absolute;
	content: "Sign Me Up";
	left: -6px;
	right: -6px;
	top: -6px;
	bottom: -10px;
	z-index: -2;
}

.push_button:active {
	top: 5px;
}

.push_button:active:before {
	top: -11px;
	bottom: -5px;
	content: "";
}

.green {
	text-shadow: -1px -1px 0 #5aaf2b;
	background: #4a9123;
	background-image: -webkit-linear-gradient(to right, #4a9123, #5aaf2b);
	background-image: -moz-linear-gradient(to right, #4a9123, #5aaf2b);
	background-image: -ms-linear-gradient(to right, #4a9123, #5aaf2b);
	background-image: -o-linear-gradient(to right, #4a9123, #5aaf2b);
	background-image: linear-gradient(to bottom, #4a9123, #5aaf2b);
	-webkit-border-radius: 5px;
	-moz-border-radius: 5px;
	border-radius: 5px;
	-moz-box-shadow: 0 1px 0 rgba(255, 255, 255, .5) inset, 0 -1px 0 rgba(255, 255, 255, .1) inset, 0 4px 0 #4a9123, 0 4px 2px rgba(0, 0, 0, .5);
	box-shadow: 0 1px 0 rgba(255, 255, 255, .5) inset, 0 -1px 0 rgba(255, 255, 255, 0.1) inset, 0 4px 0 #4a9123, 0 4px 2px rgba(0, 0, 0, .5);
}

.green:hover {
	background: #4a9123;
	background-image: -webkit-linear-gradient(to right, #5aaf2b, #4a9123);
	background-image: -moz-linear-gradient(to right, #5aaf2b, #4a9123);
	background-image: -ms-linear-gradient(to right, #5aaf2b, #4a9123);
	background-image: -o-linear-gradient(to right, #5aaf2b, #4a9123);
	background-image: linear-gradient(to right, #5aaf2b, #4a9123);
}

button.social-signin:hover, button.social-signin:focus {
	box-shadow: 0 10px 6px -6px #777;
	transition: 0.9s ease;
}

button.social-signin:active {
	box-shadow: 0 10px 6px -6px #777;
	transition: 0.2s ease;
}

button.social-signin.facebook {
	background: #32508e;
}

button.social-signin.google {
	background: #dd4b39;
}