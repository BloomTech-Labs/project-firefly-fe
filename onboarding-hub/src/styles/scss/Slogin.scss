$green: #4aa810;
$red: #e74c3c;
$blue: #3498db;
$yellow: #f1c40f;
$purple: #8e44ad;

#login-box {
	position: relative;
	margin: 5% auto;
	width: 600px;
	height: 400px;
	background: #fff;
	border-radius: 2px;
	box-shadow: 0 2px 4px rgba(0, 0, 0, 0.4);
}

.left {
	position: absolute;
	top: 0;
	left: 0;
	box-sizing: border-box;
	margin: 40px;
	width: 300px;
	height: 400px;
}

h1 {
	margin: 0 0 20px 0;
	font-weight: 300;
	font-size: 28px;
}

input[type='text'],
input[type='email'],
input[type='password'] {
	display: block;
	box-sizing: border-box;
	margin-bottom: 20px;
	padding: 4px;
	width: 220px;
	height: 32px;
	border: none;
	border-bottom: 1px solid #aaa;
	font-family: 'Roboto', sans-serif;
	font-weight: 400;
	font-size: 15px;
	transition: 0.2s ease;
}

input[type='text']:focus,
input[type='email']:focus,
input[type='password']:focus {
	border-bottom: 2px solid #16a085;
	color: #16a085;
	transition: 0.2s ease;
}

input[type='submit'] {
	margin-top: 28px;
	width: 120px;
	height: 32px;
	background: #16a085;
	border: none;
	border-radius: 2px;
	color: #fff;
	font-family: 'Roboto', sans-serif;
	font-weight: 500;
	text-transform: uppercase;
	transition: 0.1s ease;
	cursor: pointer;
}

input[type='submit']:hover,
input[type='submit']:focus {
	opacity: 0.8;
	box-shadow: 0 2px 4px rgba(0, 0, 0, 0.4);
	transition: 0.1s ease;
}

input[type='submit']:active {
	opacity: 1;
	box-shadow: 0 1px 2px rgba(0, 0, 0, 0.4);
	transition: 0.1s ease;
}

.or {
	position: absolute;
	top: 180px;
	left: 280px;
	width: 40px;
	height: 40px;
	background: #ddd;
	border-radius: 50%;
	box-shadow: 0 2px 4px rgba(0, 0, 0, 0.4);
	line-height: 40px;
	text-align: center;
}

.right {
	position: absolute;
	top: 0;
	right: 0;
	box-sizing: border-box;
	padding: 40px;
	width: 300px;
	height: 400px;
	background: url('../../assets/welcome.png');
	background-size: cover;
	background-position: center;
	border-radius: 0 2px 2px 0;
}

.right .loginwith {
	display: block;
	margin-bottom: 40px;
	font-size: 28px;
	color: #4aa810;
	text-align: center;
}

.flex {
	display: flex;
	flex-flow: column;
	justify-content: flex-end;
	align-items: flex-end;
}

.test {
	margin-bottom: -10px;
}
.loginwith {
	position: absolute;
	bottom: 0;
	margin-right: 1.6rem;
	color: green;
	text-align: center;
}

.log-footer {
	position: fixed;
	text-align: center;
	bottom: 295px;
	right: 7.8%;
	width: 100%;
}

button.social-signin {
	width: 220px;
	height: 36px;
	padding: 5px;
	margin-top: 6px;
	font-size: 0.8em;
	border: none;
	border-radius: 8px;
	color: #fff;
	font-family: 'Gravitas', sans-serif;
	font-weight: 900;
	transition: 0.2s ease;
	cursor: pointer;
}

.promise {
	font-size: 0.6em;
	font-weight: bold;
	margin-top: -1.9em;
}

.signup-submit {
	position: absolute;
	left: 15%;
	border-color: $green;
	color: #fff;
	box-shadow: 0 0 40px 40px $green inset, 0 0 0 0 $green;
	transition: all 150ms ease-in-out;

	&:hover {
		box-shadow: 0 0 10px 0 $purple inset, 0 0 10px 4px $purple;
	}
}

button.social-signin:hover,
button.social-signin:focus {
	box-shadow: 0 10px 6px -6px #777;
	transition: 0.9s ease;
}

button.social-signin:active {
	box-shadow: 0 10px 6px -6px #777;
	transition: 0.2s ease;
}

button.social-signin.facebook {
	background: #32508e;
}

button.social-signin.google {
	background: #dd4b39;
}
