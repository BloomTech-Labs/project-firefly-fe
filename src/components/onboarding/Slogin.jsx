import React, { Component } from 'react'
import axios from 'axios'
import Swal from 'sweetalert2'
import { signUpWithGoogle } from '../../firebase/firebase.utils.js'
import { doFacebookSignUp } from '../../firebase/firebase.utils.js'

export default class Slogin extends Component {
	state = {
		credentials: {
			email: '',
			password: '',
			passwordCheck: ''
		}
	}

	handleChange = e => {

		this.setState({
			credentials: {
				...this.state.credentials,
				[e.target.name]: e.target.value
			}
		})
		console.log(this.state.credentials)
	}

	onSubmit = e => {
		e.preventDefault()
		const user = { email: this.state.credentials.email, password: this.state.credentials.password }
		// console.log('submit', user)
		if (this.state.credentials.password === this.state.credentials.passwordCheck) {
			if (this.state.credentials.password.length > 5) {
				console.log("password", this.state.credentials.password)
				axios
					.post('https://infinite-meadow-87721.herokuapp.com/auth/register', user)
					.then(res => {
						console.log(res)
						Swal.fire({
							title: "Registration Confirmed",
							text: "Congrats! You have successfully registered",
							type: "success",
							showCancelButton: false,
							confirmButtonColor: "#4A9123",
							confirmButtonText: "Enjoy your stay!",
							timer: 3000
						})
						this.props.history.push('/')
					})
					.catch(err => {
						console.log(err)
					})
			} else {
				Swal.fire("Oops!", "Password must be greater than five characters")
			}
		} else {
			console.log('x')
		}
	}

	render() {
		return (
			<div id='login-box'>
				<div className='left'>
					<h1 className='signup'>SIGN UP</h1>
					<form onSubmit={this.onSubmit}>
						{/* <input type='text' name='username' placeholder='Username' /> */}
						<input
							type='email'
							name='email'
							placeholder='E-mail'
							value={this.state.credentials.email}
							onChange={this.handleChange}
							required
						/>
						<p className='promise'>
							We will never share your information with anyone!
                    </p>

						<input
							type='password'
							name='password'
							placeholder='Password'
							value={this.state.credentials.password}
							onChange={this.handleChange}
							required
						/>

						<input
							type='password'
							name='passwordCheck'
							placeholder='Retype password'
							value={this.state.credentials.passwordCheck}
							onChange={this.handleChange}
							required
						/>

						<button className='push_button green'
							type='submit'
							name='signup_submit'
							value='Sign me up'
						>
							Sign me up</button>
					</form>
					<p className='log-footer'>
						Already have an account ? <a href='./Signin'>Click Here</a>
					</p>
				</div>

				<div className='right flex'>
					<div className='test'>
						<button onClick={doFacebookSignUp} className='social-signin facebook'>
							LOG IN WITH FACEBOOK
						</button>
						<button onClick={signUpWithGoogle} className='social-signin google'>
							LOG IN WITH GOOGLE
						</button>
					</div>
					<span className='loginwith'>{/* <br /> */}</span>
				</div>
				<div className='or'>OR</div>

			</div>
		)
	}
}