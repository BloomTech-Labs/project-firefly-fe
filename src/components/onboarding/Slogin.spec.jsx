// Import react and testing library
import React from 'react'
import { render } from '@testing-library/react'
// Import renderer for snapshot
import renderer from 'react-test-renderer'

import Slogin from './Slogin'

// Test to check that the app will display when called.
describe('<Slogin />', () => {
  it('should display', () => {
    render(<Slogin/>);
  })
  // Create a snapshot
  it('matches snapshot', () => {
    const tree = renderer.create(<Slogin />); // Generates a DOM tree

    // Snapshots are a JSON representation of the DOM tree
    expect(tree.toJSON()).toMatchSnapshot();
  });
})