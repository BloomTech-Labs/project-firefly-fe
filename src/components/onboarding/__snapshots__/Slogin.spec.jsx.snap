// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Slogin /> matches snapshot 1`] = `
<div
  id="login-box"
>
  <div
    className="left"
  >
    <h1
      className="signup"
    >
      SIGN UP
    </h1>
    <form
      onSubmit={[Function]}
    >
      <input
        name="email"
        onChange={[Function]}
        placeholder="E-mail"
        required={true}
        type="email"
        value=""
      />
      <p
        className="promise"
      >
        We will never share your information with anyone!
      </p>
      <input
        name="password"
        onChange={[Function]}
        placeholder="Password"
        required={true}
        type="password"
        value=""
      />
      <input
        name="passwordCheck"
        onChange={[Function]}
        placeholder="Retype password"
        required={true}
        type="password"
        value=""
      />
      <button
        className="push_button green"
        name="signup_submit"
        type="submit"
        value="Sign me up"
      >
        Sign me up
      </button>
    </form>
    <p
      className="log-footer"
    >
      Already have an account ? 
      <a
        href="./Signin"
      >
        Click Here
      </a>
    </p>
  </div>
  <div
    className="right flex"
  >
    <div
      className="test"
    >
      <button
        className="social-signin facebook"
      >
        LOG IN WITH FACEBOOK
      </button>
      <button
        className="social-signin google"
      >
        LOG IN WITH GOOGLE
      </button>
    </div>
    <span
      className="loginwith"
    />
  </div>
  <div
    className="or"
  >
    OR
  </div>
</div>
`;
